### PR TITLE
Add helpers for hog mode

### PR DESCRIPTION
**Pull request by HEnquist**
This adds two new helpers for managing hog mode (exclusive access).

### CODE REVIEW

**Comment by HEnquist** [comment]
Does anyone have a moment to take a look at this? It's quite small with no breaking changes.

**Review by MichaelHills — COMMENTED** [review]
Seems fine to me? Will see if anyone else wants to comment, and approve in a short while if not.

**Comment by MichaelHills** [comment]
Hopefully that iOS failure should be fixed the next time you run the checks. Pretty sure I fixed/worked-around it.

**Comment by HEnquist** [comment]
Thanks for looking! I'll update asap :)

**Comment by HEnquist** [comment]
I have updated this now. I went with "toggle_hog_mode()" and "get_hogging_pid()". Not the most elegant function names ever, but I think they are at least better than my first suggestion.

File: examples/sine_advanced.rs [hunk 1]
```diff
@@ -101,7 +120,7 @@ fn main() -> Result<(), coreaudio::Error> {
     println!("setting hardware (physical) format");
     let hw_stream_format = StreamFormat {
         sample_rate: SAMPLE_RATE,
-        sample_format: SampleFormat::I24,
+        sample_format: SampleFormat::I16,
```
Comment (MichaelHills) [inline examples/sine_advanced.rs:123]: why was this changed?

**Review by MichaelHills — APPROVED** [review]
runs fine for me

File: examples/sine_advanced.rs (hunk 1 above)
Comment (HEnquist) [reply examples/sine_advanced.rs:123]: I needed to change it to run on the headphone output of my old MacBook. Then I decided to not change it back, thinking that I16 probably is a better choice. I think this should allow the example to run unmodified on more hardware.

Comment (MichaelHills) [reply examples/sine_advanced.rs:123]: Ah ok, sounds good.

**Comment by MichaelHills** [comment]
Doh looks like iOS build is failing again, after github actions upgraded to macOS 11 from 10.

**Comment by MichaelHills** [comment]
Looks like this fixes it https://github.com/RustAudio/coreaudio-rs/pull/85

**Comment by HEnquist** [comment]
Thanks for reviewing and merging! And I see you already fixed the iOS build :)
